### DISCUSSION TITLE
Could someone expand a bit on the Automatic Image Processing?

### DISCUSSION THREAD

**Discussion by darrenaddy** [Lednerb/bilberry-hugo-theme#353] [Q&A]
The README.md says (briefly):

> Automatic Image Resizing
> Bilberry theme includes built-in automatic cropping and image resizing, activated by default. However, if you want to disable it, set the resizeImages parameter to false in the config.toml file. Also, this feature can be disabled at the post level by setting the resizeImages front matter variable to false.

Could someone explain what this Automatic Image Resizing consists of and does? For example, comparing the original images in the /gallery/my-trip-to-scotland directory with those processed in the /public/gallery/my-trip-to-scotland directory - the original and the processed have different filenames, but are the SAME size (750 x 350 pixels) - and not cropped to a different aspect ratio. So, effectively, no processing was done, except you can tell it went through the process because of the newly generated filename.

Just wondering what the output breakpoints are. Is there only one breakpoint (to resize to 750px wide) and is this changeable? I've seen other responsive solutions with Hugo that resize to up to 4 different responsive breakpoints. And what is logic of the CROPPING vs the RESIZING? 

Also, is ANY image automatically cropped and/or resized (or does it apply only to images in the page bundle or posts/pages of certain content-type(s)? (article, audio, code, gallery, link, page, picture, quote, status, video). 

I searched all of the theme's code for the various Hugo Image Processing Methods.
in partials/content-type/
the following files contain .Fill: 

- gallery.html
- status.html (for featuredImage)
- featuredImage.html

layouts/_default/baseof.html uses .Fill twice (one of which is for featuredImage).

Image Processing Methods NOT USED (as far as I can tell):

- .Fit
- .Filter
- .Exif
- .Resize

THX!

**Comment by igor-baiborodine** [comment]
The documentation on image resizing has been updated: https://github.com/Lednerb/bilberry-hugo-theme#automatic-image-resizing

Here is test content for featured image resizing: https://www.bilberry-sandbox.kiroule.com/tags/featured/

>Just wondering what the output breakpoints are. Is there only one breakpoint (to resize to 750px wide), and is this changeable? 

Not, it is not changeable or configurable at the moment. It is hardcoded to `700x350`.
